examples/hmm.py

```diff
@@ -155,8 +156,11 @@ def main(args):
         transition_prior, emission_prior, supervised_categories,
         supervised_words, unsupervised_words,
     )
+    start = time.time()
     samples = mcmc(args.num_warmup, args.num_samples, init_params,
-                   potential_fn=potential_fn, constrain_fn=constrain_fn)
+                   potential_fn=potential_fn, constrain_fn=constrain_fn,
+                   progbar=True)
+    print('\nMCMC elapsed time:', time.time() - start)
```

Comment:
@neerajprad I think that `hmm` is a good example to track down performance issue because the costs in tests are just at most 2 seconds. To make it easy to address this issue, I added a condition `progbar=None` to keep the old behaviour (meaning: jit fori_loop). To my observation, with `num_warmup=500`, `num_samples=1000`, the elapsed time is 50s vs 63s. While with `num_warmup=5000`, `num_samples=10000`, the elapsed time is 115s vs 128s. This 13s is magical and consistent regardless num_warmup and num_samples (I did try `num_warmup=5/50`, `num_samples=10/100`). So I believe that something is cached with `jit(fori_loop)` but don't know which one.
(note that to make a fair benchmark, we need to disable fast math mode: `XLA_FLAGS=--xla_cpu_enable_fast_math=false`)

On the other hand, I reopen https://github.com/pyro-ppl/numpyro/issues/88 because we need to address 2-time compiling issue for `pmap`. I have a feeling that the above 13s is related to that issue so I will make an attempt to address that issue in the follow-up PR. Anyway, it is not related and things should be ready for us to discuss about the interface of `mcmc` class. Let's chat about that when you have time. :)

Reply:
It turns out (#196) that there is a huge benefit of using 1-time compiling. #196 makes the above benchmark results obsolete and only be good for reference. :)

Reply:
@fehiepsi - If this 13 s. is mostly constant and doesn't scale with number of samples, we can remove `jit`. 

> ready for us to discuss about the interface of mcmc class.

What's the change to the interface that you were thinking of?

Reply:
@neerajprad I was worried that I might make some wrong adjustment, so I keep the old behaviour so you can easily try on other examples/benchmarks. :)

> What's the change to the interface that you were thinking of?

Let's discuss some of them in #196. For `mcmc`, I think that we just need 2 flags `num_chains` and `parallel=True/False`. But we need some changes in `summary` and `fori_collect`.

Reply:
Btw, unable to use `jit(fori_loop)` is a bug and has been fixed upstream. So we can use `jit(fori_loop)` now. Sorry for driving the implementation back and forth! Anyway, I think this PR is still good for having `pmap` tested before the next jax release. :)